adareducer/engine.py

```diff
@@ -194,14 +219,31 @@ def next_file_to_process(self):
     def run(self):
         """Run self: reduce the project as much as possible"""
 
+        all_files = set([os.path.abspath(f) for f in os.listdir(self.project_dir)])
```

Comment:
Now that we have Proper project support, let's do something different to get the full list: we can either only focus on trying to delete all project source files, or we could go one step further (as you're doing right now) and look at all files in all source dirs of the project tree. As you wish.

The object self.resolver is a ProjectResolver (see project_support.py): this object should be augmented to give us the list of source dirs if we want to go that route.

Reply:
Note that I have a slight preference for attempting project sources only, this saves us from having to be careful about predicate artifacts or adareducer artifacts, while still being useful for the majority of use case.

Reply:
From a user perspective I think that I would prefer to have something that cleans up as many files as possible, leaving me with something that I could directly file in my testsuite.

But I now realize that this is in fact very dangerous and that this could remove files in, e.g. `.git`, which is certainly something we don't want. So I'll follow your advice.